app/uk/gov/hmrc/cbcr/repositories/ReportingEntityDataRepo.scala

```diff
@@ -310,4 +343,55 @@ class ReportingEntityDataRepo @Inject()(protected val mongo: ReactiveMongoApi)(i
     } yield update.nModified
   }
 
+  def datesAreOverlapping(existingData: List[ReportingEntityData], entityReportingPeriod: EntityReportingPeriod) = {
+    val groupedData =
+      existingData
+        .filter(_.reportingPeriod.isDefined)
+        .groupBy(_.reportingPeriod.get)
+    val res = groupedData.map(data => getLatestReportingEntityData(data._2)).filterNot(_.isEmpty).map(_.head)
+
+    val filteredDeletionsAndSamePeriod =
+      res.filter(data => filterOutDeletion(data)).filter(p => p.reportingPeriod.get != entityReportingPeriod.endDate)
+
+    //mainly for backward compatibility make sure reporting period doesn't overlap with the new submission
+    //true = no overlapping
+    val firstCheck: Boolean =
+      filteredDeletionsAndSamePeriod.forall(d => !checkBySingleDate(entityReportingPeriod, d.reportingPeriod.get))
+
+    val secondCheckList = filteredDeletionsAndSamePeriod.filter(_.entityReportingPeriod.isDefined)
+
+    //Make sure when we have both dates that they don't overlap true = no overlapping
+    val secondCheck: Boolean =
+      secondCheckList.forall(d => !checkBothDates(entityReportingPeriod, d.entityReportingPeriod.get))
+
+    if (firstCheck && secondCheck) false else true
```

Comment:
Could just shorten to !(firstCheck && secondCheck)

Reply:
Done